client/me/purchases/add-new-payment-method/index.jsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StripeHookProvider } from '@automattic/calypso-stripe';
+
+/**
+ * Internal dependencies
+ */
+import { addStoredCard } from 'calypso/state/stored-cards/actions';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { concatTitle } from 'calypso/lib/react-helpers';
+import { createCardToken, getStripeConfiguration } from 'calypso/lib/store-transactions';
+import PaymentMethodForm from 'calypso/me/purchases/components/payment-method-form';
+import DocumentHead from 'calypso/components/data/document-head';
+import FormattedHeader from 'calypso/components/formatted-header';
+import HeaderCake from 'calypso/components/header-cake';
+import Main from 'calypso/components/main';
+import titles from 'calypso/me/purchases/titles';
+import { paymentMethods } from 'calypso/me/purchases/paths';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import Layout from 'calypso/components/layout';
+import Column from 'calypso/components/layout/column';
+import PaymentMethodSidebar from 'calypso/me/purchases/components/payment-method-sidebar';
+
+function AddNewPaymentMethod( props ) {
+	const createAddCardToken = ( ...args ) => createCardToken( 'card_add', ...args );
```

Comment:
This, too, presents a risk because the original component is being changed in https://github.com/Automattic/wp-calypso/pull/47642, although it's not as obvious how to prevent the duplication in this case.

Reply:
hmm, yeah that's tricky. One thought that comes to mind is to extra, and centralize the parts of this that both the components would need. Does that make sense?